nearcore/src/state_sync.rs

```diff
@@ -492,51 +466,34 @@ fn obtain_and_store_state_part(
     Ok(state_part)
 }
 
-/// Gets basic information about the epoch to be dumped.
-fn start_dumping(
-    epoch_id: EpochId,
-    sync_hash: CryptoHash,
-    shard_id: ShardId,
+fn cares_about_shard(
     chain: &Chain,
-    epoch_manager: &dyn EpochManagerAdapter,
+    shard_id: &ShardId,
+    sync_hash: &CryptoHash,
     shard_tracker: &ShardTracker,
     account_id: &Option<AccountId>,
-) -> Result<Option<StateSyncDumpProgress>, Error> {
-    let epoch_info = epoch_manager.get_epoch_info(&epoch_id)?;
-    let epoch_height = epoch_info.epoch_height();
-
+) -> Result<bool, Error> {
     let sync_header = chain.get_block_header(&sync_hash)?;
     let sync_prev_hash = sync_header.prev_hash();
-    let sync_prev_header = chain.get_block_header(&sync_prev_hash)?;
-    // Need to check if the completed epoch had a shard this account cares about.
-    // sync_hash is the first block of the next epoch.
-    // `cares_about_shard()` accepts `parent_hash`, therefore we need prev-prev-hash,
-    // and its next-hash will be prev-hash. That is the last block of the completed epoch,
-    // which is what we wanted.
-    let sync_prev_prev_hash = sync_prev_header.prev_hash();
-
-    let state_header = chain.get_state_response_header(shard_id, sync_hash)?;
-    let num_parts = get_num_state_parts(state_header.state_root_node().memory_usage);
-    if shard_tracker.care_about_shard(account_id.as_ref(), sync_prev_prev_hash, shard_id, true) {
-        tracing::info!(target: "state_sync_dump", shard_id, ?epoch_id, epoch_height, %sync_prev_hash, %sync_hash, "Dumping state of Epoch");
-        // Note that first the state of the state machines gets changes to
-        // `InProgress` and it starts dumping state after a short interval.
-        set_metrics(&shard_id, Some(0), Some(num_parts), Some(epoch_height));
-        Ok(Some(StateSyncDumpProgress::InProgress { epoch_id, epoch_height, sync_hash }))
-    } else {
-        tracing::info!(target: "state_sync_dump", shard_id, ?epoch_id, epoch_height, %sync_hash, "Shard is not tracked, skip the epoch");
-        Ok(Some(StateSyncDumpProgress::AllDumped { epoch_id, epoch_height }))
-    }
+    Ok(shard_tracker.care_about_shard(account_id.as_ref(), sync_prev_hash, *shard_id, true))
```

Comment:
one more thing, should this be sync_prev_hash, or sync_prev_prev_hash?
This was the comment in the old old code `// `cares_about_shard()` accepts `parent_hash`, therefore we need prev-prev-hash,`

Reply:
but this does not seem to explain why your node was not dumping the latest epoch with old code...

Reply:
sync_prev_hash and sync_prev_prev_hash are in the same epoch. Getting sync_prev_hash is easier by a few lines.